compiler/ballerina-lang/src/main/java/io/ballerina/projects/internal/BlendedManifest.java

```diff
@@ -69,8 +69,20 @@ public static BlendedManifest from(DependencyManifest dependencyManifest,
                     depInPkgManifest.repository().equals(ProjectConstants.LOCAL_REPOSITORY_NAME) ?
                     Repository.LOCAL : Repository.NOT_SPECIFIED;
 
-            if (!localPackageRepository.isPackageExists(depInPkgManifest.org(), depInPkgManifest.name(),
-                                                       depInPkgManifest.version())) {
+            if (depInPkgManifest.repository() != null &&
+                    depInPkgManifest.repository().equals(ProjectConstants.LOCAL_REPOSITORY_NAME)) {
+                if (!localPackageRepository.isPackageExists(depInPkgManifest.org(), depInPkgManifest.name(),
+                        depInPkgManifest.version())) {
+                    continue;
+                }
+            } else {
+                depContainer.add(depInPkgManifest.org(), depInPkgManifest.name(),new Dependency(
+                        depInPkgManifest.org(), depInPkgManifest.name(), depInPkgManifest.version(),
+                        DependencyRelation.UNKNOWN, Repository.NOT_SPECIFIED,
+                        moduleNames(new DependencyManifest.Package(depInPkgManifest.name(), depInPkgManifest.org(),
+                                depInPkgManifest.version())),DependencyOrigin.LOCKED)
```

Comment:
The dependency origin is `DependencyOrigin.USER_SPECIFIED`

Reply:
Changed to `USER_SPECIFIED`